src/NLog/Internal/StringBuilderExt.cs

```diff
@@ -113,26 +119,73 @@ public static void AppendInvariant(this StringBuilder builder, uint value)
                 length++;
             }
             while (length_calc > 0);
+            return length;
+        }
 
+        private static void ApppendValueWithDigitCount(StringBuilder builder, uint value, int digitCount)
+        {
             // Pad out space for writing.
-            builder.Append('0', length);
+            builder.Append('0', digitCount);
 
             int strpos = builder.Length;
 
             // We're writing backwards, one character at a time.
-            while (length > 0)
+            while (digitCount > 0)
             {
                 strpos--;
 
                 // Lookup from static char array, to cover hex values too
                 builder[strpos] = charToInt[value % 10];
 
                 value /= 10;
-                length--;
+                digitCount--;
             }
         }
+
         private static readonly char[] charToInt = new char[] { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' };
 
+        /// <summary>
+        /// Convert DateTime into UTC and format to yyyy-MM-ddTHH:mm:ss.fffffffZ - DateTime.ToString("o")
```

Comment:
Maybe note that this will render a ISO 6801 date?

Reply:
Fixed